### PR TITLE
Add documentation on the Swerve Generator

### PR DESCRIPTION
**Pull request by daltzctr**
Depends on #93

### CODE REVIEW

File: source/docs/tuner/tuner-swerve/swerve-system-requirements.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+          <input type="checkbox"/>
+
+   * - **RECOMMENDED:** Have devices been renamed?
+
+       (e.g. "TalonFX ID 1 -> FL Steer Motor")
+
+     - .. raw:: html
+
+          <input type="checkbox"/>
```
Comment (CoryNessCTR) [inline source/docs/tuner/tuner-swerve/swerve-system-requirements.rst:73]: Duplicate of above

File: source/docs/tuner/tuner-swerve/swerve-system-requirements.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  * Firmware version should begin with 24.X
+  * Ensure Tuner shows server version as "2024" or newer
+
+* Temporary diagnostic server (or an existing 2024 robot project) must be running and Tuner should be connected to the robot.
+
+  * This allows the generator to perform it's setup and auto-calibration routines
+
+Some of these requirements are inforced via a mechanism called "precheck". You can see and refresh the precheck status by clicking the :guilabel:`Refresh` button in the top right.
+
```
Comment (CoryNessCTR) [inline source/docs/tuner/tuner-swerve/swerve-system-requirements.rst:22]: Maybe include a "recommended" section for CANivore use, Pro license, and the device renaming.

File: source/docs/tuner/tuner-swerve/validating-drivetrain.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+This test consists of two steps:
+
+* Verify Drive
+* Verify Steer
+
+.. image:: images/swerve-verification-page.png
+   :alt: Picture of the verification page in Tuner
+
+Verify Drive
```
Comment (CoryNessCTR) [inline source/docs/tuner/tuner-swerve/validating-drivetrain.rst:14]: We should make it verify steer before the drive first in Tuner, and the documentation should be steer first, drive second to match.

File: source/docs/tuner/tuner-swerve/swerve-system-requirements.rst (hunk 2 above)
Comment (daltzctr) [reply source/docs/tuner/tuner-swerve/swerve-system-requirements.rst:22]: I think the recommendations in the requirements checklist will suffice. I'll add licensing to the recommended requirements there.

File: source/docs/tuner/tuner-swerve/swerve-system-requirements.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
+Swerve Requirements
+===================
+
+The swerve project creator and :doc:`swerve API </docs/tuner/tuner-swerve/index>` have several limitations. These limitations are in-place to maximize performance and improve maintainability.
```
Comment (CoryNessCTR) [inline source/docs/tuner/tuner-swerve/swerve-system-requirements.rst:4]:
```suggestion
The swerve project creator and :doc:`swerve API </docs/tuner/tuner-swerve/index>` have several limitations. These limitations are in place to maximize performance and improve maintainability.
```

File: source/docs/tuner/tuner-swerve/configuring-your-modules.rst [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+-------------------
+
+Once a module's devices have been selected, an :guilabel:`Encoder Calibration` button will appear on the right-side of the screen. This will open a popup that guides the user through aligning their module and calibrating their CANcoder offsets.
+
+.. important:: It's extremely important for the modules to be aligned such that the bevel gear faces the **vertical center** of the robot. Failure to perform this step may lead the drive verification tests to fail.
+
+.. image:: images/swerve-encoder-calibration-loc.png
+   :alt: Red bounding box around the swerve calibration button in Tuner X
+
+Once calibration has been completed, the module will be removed from the **Incomplete Modules** lift on the left.
```
Comment (CoryNessCTR) [inline source/docs/tuner/tuner-swerve/configuring-your-modules.rst:31]:
```suggestion
Once calibration has been completed, the module will be removed from the **Incomplete Modules** list on the left.
```

File: source/docs/tuner/tuner-swerve/validating-drivetrain.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+Verify Steer
+------------
+
+.. image:: images/swerve-verify-steer.png
+   :alt: Picture of the steer verification page in Tuner
+
+The **Verify Steer** test is used to confirm that the module azimuth (or steer) can rotate freely and what direction they rotate in.
+
+This test simply rotates all of the modules and confirms with the user if they rotated clockwise or counter-clockwise. The modules should rotate **counter-clockwise**.
```
Comment (CoryNessCTR) [inline source/docs/tuner/tuner-swerve/validating-drivetrain.rst:22]: Clockwise or counterclockwise as viewed from what direction?
If I'm looking at it from the bottom, that's different than looking at it from the top.

**Review by CoryNessCTR — CHANGES_REQUESTED** [review]

**Review by CoryNessCTR — APPROVED** [review]

**Pull request merged**